marketdata/stream/client.go

```diff
@@ -417,6 +427,17 @@ func (c *client) maintainConnection(ctx context.Context, u url.URL, initialResul
 				continue
 			}
 			c.logger.Infof("datav2stream: finished connection setup")
+
+			if c.connectCallback != nil {
+				callbackWaitGroup.Add(1)
+				go func() {
+					defer callbackWaitGroup.Done()
+					if err := c.connectCallback(); err != nil {
+						c.logger.Warnf("datav2stream: connect callback failed, error: %v", err)
```

Comment:
If we all do with the returned error of the callback functions is logging then I think we shouldn't do that. They shouldn't return an error and any error handling should be done by the user inside the functions.